meetings/2022-12-05.md

```diff
@@ -0,0 +1,91 @@
+* ...: The last part is about agent registration, where agent can request a agent specific document at the discovery hub.
+* ...: There is a sequence diagram which follows all the steps.
+
+* Pavlik: Question about registration. The way I implemented it - the owner of Authz Agent handles it - the client redirects to the Authz Agent which handles it. So some user interaction is required.
+* WT: Need a way to register agents but don't stipulate the methods to do so. This spec just indicates the possibility. The basic registration proposed in the doc is to illustrate how back and forth could work but leaves implementation open to which type of registration is stipulated.
+* Pavlik: How in practice would you see it with a PR? Extract parts of SAI and then SAI extends more specifically?
+* WT: I tried to mention / use the SAI vocab in here - so parts of SAI that are similar could be left out, and then decide whether to change vocab here to reflect original vocab. Don't think it's necessary to have general and more specific - just make one that's standalone and in rest of SAI extend with specific implementation of registration procedures
+* Pavlik: Currently in Sai the Authz Agent handles discovery of Agent Registration. It has logic based on client and end user identities to return the right registration. Would you see this part defined here or in SAI
+* WT: Could include here, or could be implementation direction
+* Pavlik: Discussing possibility of having a special public agent registration and IRI for public agent (I think ACP does this). If we have something like that - maybe there would be two link headers returned for public and agent specific - or add a special triple to shortcut the hub - because you won't need to request authorization for public agent
+* WT: If you're a public agent and you act as such - and are registered nowhere - then you could nevertheless pull from the RDH to get the public agent location, or could use a specific triple in the webid. But using this mechanism you can separate technical stuff from the webid and leave it open for read/write of profile or whatever else you want to do with it.
+* Pavlik: I wouldn't say the client is acting as a public agent - any agent is always acting as a public agent plus an authorized agent. So you can get agent specific one plus public one. If you wanted one entry point it would be useful to always get the public registration as a separate link in a header. Advertising in a webid document could be seen as a shortcut (nice-to-have). Could also have a different predicate to advertise it. Would still have a structure of agent registration. To create it you'd still want to use an authorization agent.
```

Comment:
```suggestion
* Pavlik: I wouldn't say the client is acting as a public agent — any agent is always acting as a public agent plus an authorized agent. So you can get agent-specific one plus public one. If you wanted one entry point, it would be useful to always get the public registration as a separate link in a header. Advertising in a WebID document could be seen as a shortcut (nice-to-have). Could also have a different predicate to advertise it. Would still have a structure of agent registration. To create it, you'd still want to use an authorization agent.
```